src/com/tictacgo/TicTacGoGame.java

```diff
@@ -0,0 +1,402 @@
+package com.tictacgo;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+import android.widget.PopupWindow;
+import android.widget.TableRow;
+import android.widget.TextView;
+
+import com.tictacgo.data.Board;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class TicTacGoGame extends Activity {
+  /**
+   * The Board of the game
+   */
+  private Board board;
+
+  /**
+   * The FrameLayout in which the game is to be drawn
+   */
+  private FrameLayout fl;
+
+  /**
+   * OnClickListener to use for each piece
+   */
+  private View.OnClickListener pieceClicked;
+
+  /**
+   * OnClickListener to use to pick direction
+   */
+  private View.OnClickListener directionClicked;
+
+  /**
+   * PopupWindow with direction Buttons
+   */
+  private PopupWindow directions;
+
+  /**
+   * The height of the usable screen area
+   */
+  private int height;
+
+  /**
+   * An ArrayList of the Boards for undo and redo
+   */
+  private List<Board> undoHistory;
+
+  /**
+   * The index of the current undo/redo history
+   */
+  private int historyIndex;
+
+  /**
+   * The initial turn selection. Used for the New Game Button
+   */
+  private Board.Player turn;
```

Comment:
Might want to import Board.Player separately so you can just call these Players.